### DISCUSSION TITLE
Getting Sveltekit to display some data from Strapi via API

### DISCUSSION THREAD

**Discussion by NafunaAfrica** [sveltejs/kit#9684] [Q&A]
Hello all! I do hope you will help me with this issue I am facing.

So I have a web app built on Sveltekit. Here I intend to pull data from strapi for a blog as well as other information. I have made a route called news and a nested route called posts which is my API endpoint. Here in '/news/posts' I check with Postman and the demo data I have there flows in well:
![image](https://user-images.githubusercontent.com/84924809/232476617-c7041a95-737c-4e3c-ac16-1baf7c67350b.png)
The code that did that in the +server.ts files is:

`   export const GET = async ({ request }) => {

   const res = await fetch('https://cms.nafuna.tv/api/blogs')
   const info = await res.json()

    return new Response(JSON.stringify(info), { status: 200})
}`

so now I wanted to get the data into the main +page.svelte file. I have made a +page.server.ts to load the data from the endpoint:

`export async function load({ fetch }) {
    const res = await fetch('/news/posts')
    const info = await res.json()
    return { props: { blogs: info.data } };
}
`

My main issue happens when I am trying to use that in the +page.svelte whose code looks like:

`<script lang="ts">
    import type { PageData } from './$types';
    
    export let data;
    
 
</script>
{#each blogs as post}
<h1>{post.Title}</h1>
{/each}
`
the above won't display the posts coming from API. What am I missing? I'm pretty new to Sveltekit and have been learning Javascript for under 3 months so there maybe things that I know I missed. Here is a screenshot side by side so you can better see:

![image](https://user-images.githubusercontent.com/84924809/232478014-2e65a696-c298-469c-a9ef-8ba6c09d3445.png)

Kind regards!

**Comment by Glench** [comment]
I'm not sure but I think the error might be in `+page.server.ts`. Shouldn't the last line be this?

```ts
return { props: { blogs: info.data } }
// not `res.data`
```

**Reply by NafunaAfrica** [reply]
Hi there! Thank you for that. I have changed it to that before as I was trying and it had failed. I've tried again so I can share the errors:
![image](https://user-images.githubusercontent.com/84924809/232580149-adafb7c1-a9a8-449c-8559-3a85f3841534.png)

**Reply by Glench** [reply]
Aha, but now shouldn't your +page.svelte code be this?

```svelte
{#each data.props.blogs as post}
```

**Reply by NafunaAfrica** [reply]
Yes, thank you! I managed to get rid of the errors in the code as the squiggly red lines are gone BUT:

![image](https://user-images.githubusercontent.com/84924809/232667098-4ac36c32-ce23-4cfb-98e7-6bccd6f2c1b7.png)


Now I get the above instead of the content I have in strapi.

**Comment by NafunaAfrica** [comment]
![image](https://user-images.githubusercontent.com/84924809/232731653-bdb4c005-6c4a-4efd-a0a5-8975f6825244.png)

thats awesome! Progress but this is what I get versus the posts when I add one more into strapi it shows up, so the undefined are exactly like the posts I have on the cms

**Comment by NafunaAfrica** [comment]
Additional console logs and errors:
![image](https://user-images.githubusercontent.com/84924809/232742465-70ee9d9a-5d85-46f2-81fc-716c6adf4e4f.png)

here is one from Chrome. I logged out 'data' on the page.svelte and I can now see the data coming in:

![image](https://user-images.githubusercontent.com/84924809/232743705-44f07c54-2782-46ab-85df-2c290941872f.png)


not sure why it wont print?

**Reply by Glench** [reply]
Nice progress!

From what you printed out with your `console.log` command, it looks like each post object doesn't have a `post.Title` as shown in your code. I can't quite see because of the `...` in your screenshot but I'm guessing it's something like `post.attributes.Title`.

**Reply by NafunaAfrica** [reply]
And that's exactly what it was! I just needed to put the attributes in there and it printed!

![image](https://user-images.githubusercontent.com/84924809/232759215-4f584b5c-6e52-438e-97fb-96321f8baf95.png)

I am well on my way! Thank you so much for all the help! One last aspect to this is that the images don't yet load as they came in as a string of text. How to I parse the strapi rich text into my Sveltekit app?

Also there was a featured image content item which is to have a featured image, that's not showing up somehow as well....

Thank you again for all the help! I appreciate it!

**Reply by Glench** [reply]
Yay! You're welcome. I don't know anything about Strapi but it looks like it gives you markdown-formatted text, so you'll need to find a Svelte plugin that renders markdown for you. I've actually never done this myself but I imagine it's really easy to find a library that does it for you.

**Comment by NafunaAfrica** [comment]
Hello again. Thought to pick this up from here versus making a new thread. 

My main new challenge now is setting up a system where the user clicks on the blog item and then they see a bigger page with the full article. I know this is a slug system and I have made a route called: '/news/[slug]' and in there I have 2 files, a page.svelte and a page.server.ts. I copy pasted the working API call code into the new page.ts but not too sure how to bring all the data in without using a for each loop.

totally new to dynamic routes. I have shower created a slug item on the strapi side.

The new server.ts code is:
`export async function load({ params, fetch }) {
    const res = await fetch(`https://cms.nafuna.tv/api/blogs/${params.id}?populate=*`);
    const info = await res.json()
    return { props: { blogs: info.data.attributes } };
    console.log(props);
}
`

**Reply by Glench** [reply]
Do you know how to query your API for a single post instead of all of them?

**Reply by NafunaAfrica** [reply]
Hello again @Glench!

So I have settup slug option in strapi:
![image](https://user-images.githubusercontent.com/84924809/233811794-b6b92182-99d3-4aa2-a997-771c8e560db6.png)

It hyphenated the title but still checking how to query

**Reply by Glench** [reply]
Looks like you can use this api endpoint to query individual blog articles: https://docs.strapi.io/dev-docs/api/rest#get-an-entry

And maybe query by slug: https://forum.strapi.io/t/strapi-v4-search-by-slug-instead-id/13469/3
